cmd/team-member-updater/main.go

```diff
@@ -0,0 +1,230 @@
+			} else {
+				githubIDsbyGroup[member.group].Insert(member.githubID)
+			}
+		}
+
+		if member.team != "" {
+			if githubIDsbyTeam[member.team] == nil {
+				githubIDsbyTeam[member.team] = sets.NewString(member.githubID)
+			} else {
+				githubIDsbyTeam[member.team].Insert(member.githubID)
+			}
+			// maintainer teams are needed for granular GitHub permissions, leads are maintainers
+			if sets.NewString(roleTeamLead, roleMaintainer).Has(member.role) {
+				teamName := fmt.Sprintf("%s Maintainers", member.team)
+				if githubIDsbyTeam[teamName] == nil {
+					githubIDsbyTeam[teamName] = sets.NewString(member.githubID)
+				} else {
+					githubIDsbyTeam[teamName].Insert(member.githubID)
+				}
+			}
+		}
+		// only a couple roles are special
+		if sets.NewString(roleArchitect, roleGroupLead, roleTeamLead).Has(member.role) {
+			if githubIDsbyRole[member.role] == nil {
+				githubIDsbyRole[member.role] = sets.NewString(member.githubID)
+			} else {
+				githubIDsbyRole[member.role].Insert(member.githubID)
+			}
+		}
+	}
+
+	updateTeamMembers := func(name, description string, members []string) {
+		team, exists := orgConfig.Orgs["openshift"].Teams[name]
+		if !exists {
+			team = org.Team{
+				TeamMetadata: org.TeamMetadata{
+					Description: &description,
+				},
+				Maintainers: []string{"openshift-ci-robot"},
+			}
+		}
+		team.Members = members
+		orgConfig.Orgs["openshift"].Teams[name] = team
+	}
+
+	for group, ids := range githubIDsbyGroup {
+		name := fmt.Sprintf("OpenShift Group %s", group)
+		description := fmt.Sprintf("Members of the OpenShift development group %s", group)
+		updateTeamMembers(name, description, ids.List())
+	}
+
+	for team, ids := range githubIDsbyTeam {
+		name := fmt.Sprintf("OpenShift Team %s", team)
+		description := fmt.Sprintf("Members of the OpenShift development team %s", team)
+		updateTeamMembers(name, description, ids.List())
+	}
+
+	for role, ids := range githubIDsbyRole {
+		name := fmt.Sprintf("OpenShift %ss", role)
+		description := name
+		updateTeamMembers(name, description, ids.List())
+	}
+
+	org := orgConfig.Orgs["openshift"]
+	org.Members = allGithubIDs.Difference(sets.NewString(org.Admins...)).List()
+	orgConfig.Orgs["openshift"] = org
+
+	edited, err := yaml.Marshal(orgConfig)
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to marshal edited org config.")
+	}
+
+	if err := ioutil.WriteFile(o.configPath, edited, 0666); err != nil {
+		logrus.WithError(err).Fatal("Failed to write edited org config.")
+	}
+	logrus.Info("Updated org config using spreadsheet data.")
+}
+
+// teamMemberFromRow parses a spreadsheet row into a record; columns are:
+//   0: empty
+//   1: group
+//   2: team
+//   3: time
+//   4: empty
+//   5: empty
+//   6: name
+//   7: empty
+//   8: time zone
+//   9: empty
+//  10: role
+//  11: empty
+//  12: notes
+//  13: empty
+//  14: valid team
+//  15: email
+//  16: github id
+func teamMemberFromRow(row []string) teamMember {
```

Comment:
This assumes/trust that the `row` will be always a specific format. Can we validate this here and return an error before trying to access a nil index?